src/api/client.ts

```diff
@@ -144,3 +145,34 @@ const shouldSkipLogging = (error: AxiosError) => {
     (error.config?.['axios-retry'] as any)?.retryCount < RETRY_COUNT;
   return configuredToSkipLogging || willRetry;
 };
+
+export type TimeoutRequest = {
+  didTimeout: boolean;
+  signal: AbortSignal;
+  start(): void;
+  clear(): void;
+  abort(): void;
+};
+
+export const useTimeoutRequest = (): TimeoutRequest => {
+  const controller = new AbortController();
+  var didTimeout = false;
+  var timerId: NodeJS.Timeout | undefined;
+
+  const start = () => {
+    timerId = setTimeout(() => {
+      didTimeout = true;
+      controller.abort();
+    }, DEFAULT_TIMEOUT);
+  };
+
+  return {
+    didTimeout,
+    signal: controller.signal,
+    start,
+    clear: () => {
+      timerId && clearTimeout(timerId);
+    },
+    abort: () => controller.abort(),
+  };
+};
```

Comment:
To quickly answer yes I tried, but that doesn't work as expected, it fails or is broken for react-native in major cases. I found many talks about it for example https://github.com/axios/axios/issues/647: it is a post from 2017 and still a thing these days. Not sure if it was well tested when you guys started the app but, today it is not behaving as expected. So after an exhausting test, I saw it wasn't working at all and the loading spin ran forever. So I saw many devs were moving to `axios.CancelToken.source()`(Deprecated now, AbortController is the new way of doing it) and I wanted to give it a try! and voila! It works as it should! plus to that now we can cancel the request as far as the user does not need the response anymore on the `useEffect`, those calls were hanging in the background without use.
This implementation which is only partially implemented on the departures calls is working properly and the timeout is taken into consideration when the server delays.

Reply:
Great research! That's why when testing `timeout` earlier it didn't work when I used fly-mode on my phone.

I think it should be possible to encapsulate this more in the `client.ts`? 🤔 I think there is some unreleased potential there. It seems that clearing the timeout should be possible to be handled automatically by the client?

I think something like this would be good from the user side:
```ts
// Timeout, clearing timer, etc. handled by `client.ts`. 
// Just necessary to create `AbortController` on the user side.
useEffect(() => {
  const abortController = new AbortController();
  client.fetchTickets({abortController, timeout: 15000}).then(setTickets);
  return () => abortController.abort();
}
```

Or maybe even something like this:
```ts
// Here also creating the abort controller is done in `client.ts`. Then both
// abort implementation and timeout implementation is encapsulated away
// from the user side.
useEffect(() => {
  const {abort, run} = client.fetchTickets({timeout: 15000});
  run().then(setTickets);
  return abort; // Same as 'return () => abort()'
}
```

Reply:
Yes! Agree!, the intention I had, having it inside `client.ts` is to inject this in a more generic function for each request, so instead of calling `client.get` or `client.post` call a generic request:

```
export default function axiosRequest(config: any) {
   // Timeout implementation ...
   return {
        run: async () => await client.axios(config),
        abort,
        etc,
    }
}
```
But at the same time, doing this I could end up changing all the requests of the app, and increasing the size of the task, but I agree that it should be more generic for all other calls.

The app uses a concept of `reducer with side effects` which internally calls a request, so there are 3 deep levels here because it is needed at the `useEffect` level call abort as well inside the reducer.

For now, I just implemented it in the most generic way, which should work for the task. However, if encapsulating it (as your second suggestion) is straightforward forward just let's do it.

Reply:
Approved! I think this can be done more generic, but I agree, let's not dive into it now.